File: CodeForces/1-100/58/A/58A.py
# Problem 58A: Chat room
# https://codeforces.com/contest/58/problem/A
# Executed with: Python 3.11.2
# Linux version: Linux kernel 6.1.0-10-amd64
# Submitted on: September 26th, 2024
# Codeforces language used: PyPy 3.10 (7.3.15, 64bit)

string = str(input())
hello_string = "hello"
looking_for = 0

for character in string:
    if character == hello_string[looking_for]:
        looking_for += 1
    
    if looking_for == 5:
        break

if looking_for == 5:
    print("YES")
else:
    print("NO")

File: CodeForces/1-100/58/A/58A.rb
# Problem 58A: Chat room
# https://codeforces.com/contest/58/problem/A
# Executed with: ruby 3.3.8 (2025-04-09 revision b200bad6cd) [aarch64-linux-gnu]
# Linux version: Linux kernel 6.17.0-5-generic
# Submitted on: October 26th, 2025
# Codeforces language used: Ruby 3.2.2

input_str = gets.chomp
hello_str = "hello"

for index in 0..input_str.length-1
  if input_str[index] == hello_str[0]
    hello_str = hello_str[1..-1]
    if hello_str.length == 0
      puts "YES"
      return
    end
  end
end

puts "NO"
